shop_stripe/payment.py

```diff
@@ -52,6 +57,31 @@ def charge(self, cart, request):
             msg = "Stripe returned status '{status}' for id: {id}"
             raise stripe.error.InvalidRequestError(msg.format(**charge))
 
+    def subscribe(self, cart, request):
+        """
+        Create a new customer and subscribe the customer
+        to the default payment plan.
+        """
+        user_id = cart.customer.user_id;
+        user = User.objects.get(id=user_id)
+
+        try:
+            customer = StripeCustomer.objects.get(user_id=user_id)
+            stripe_customer = stripe.Customer.retrieve(customer.stripe_customer_id)
+        except ObjectDoesNotExist:
```

Comment:
... honestly I've used try/except in this way.... so I'm probably calling the kettle black at best....  but I shouldn't be your bar for quality.